src/hotspot/share/memory/metaspaceShared.cpp

```diff
@@ -122,7 +123,31 @@ char* MetaspaceShared::symbol_space_alloc(size_t num_bytes) {
   return _symbol_region.allocate(num_bytes);
 }
 
-size_t MetaspaceShared::reserved_space_alignment() { return os::vm_allocation_granularity(); }
+void MetaspaceShared::init_alignments() {
+  assert(_core_region_alignment == 0, "call this only once");
+  if (DumpSharedSpaces) {
+    _core_region_alignment = (size_t)os::vm_allocation_granularity();
+#if (defined(LINUX) && defined(AARCH64)) || (defined(__APPLE__) && defined(AMD64))
+    // If a CDS archive is created on one machine, and used on another, and the two
+    // machines have different page sizes, make sure the archive can be used on
+    // both machines.
+    //
+    // (a) Linux/aarch64 can be configured to have either 4KB or 64KB page sizes.
+    // (b) macOS/x64 uses 4KB, but macOS/aarch64 uses 64KB (note: you can run a x64 JDK
+    //     on a M1-based MacBook using Rosetta).
```

Comment:
I think the page size on Apple silicon Macs is 16KB not 64KB:

```
$ sysctl vm.pagesize
vm.pagesize: 16384
```

This is also the value reported by `sun.misc.Unsafe.pageSize()` with a native JDK. Although some Linux distributions use a 64KB page size so rounding up is probably the right thing to do.

Reply:
@nick-arm You are right --- on M1 it is 16384. Some linux-aarch64 configured 64K page size.